src/components/product/product.service.ts

```diff
@@ -63,63 +74,116 @@ export class ProductService {
+      createdAt: result.createdAt,
+      type: result.type,
+      books: result.books,
+      mediums: result.mediums,
+      purposes: result.purposes,
+      approach: result.approach,
+      methodology: result.methodology,
     };
   }
 
   async list(
     { page, count, sort, order, filter }: ProductListInput,
     session: ISession
   ): Promise<ProductListOutput> {
-    const result = await this.db.list<Product>({
-      session,
-      nodevar: 'product',
-      aclReadProp: 'canReadProducts',
-      aclEditProp: 'canCreateProduct',
-      props: [
-        { name: 'type', secure: false },
-        { name: 'books', secure: false, list: true },
-        { name: 'mediums', secure: false, list: true },
-        { name: 'purposes', secure: false, list: true },
-        { name: 'approach', secure: false },
-        { name: 'methodology', secure: false },
-      ],
-      input: {
-        page,
+    const { engagementId } = filter;
+    let result: {
+      items: Product[];
+      hasMore: boolean;
+      total: number;
+    } = { items: [], hasMore: false, total: 0 };
+
+    if (engagementId) {
+      const qry = `
+        MATCH
+          (token:Token {active: true, value: $token})
+          <-[:token {active: true}]-
+          (requestingUser:User {
+            active: true,
+            id: $requestingUserId
+          }),
+          (engagement {id: $engagementId, active: true, owningOrgId: $owningOrgId})
+          ,(product:Product {active:true})
+        WITH COUNT(product) as total, engagement, product
+          RETURN total, product.id as id, product.createdAt as createdAt
+          ORDER BY ${sort} ${order}
+          SKIP $skip LIMIT $count
+      `;
+      const productQuery = this.db.query().raw(qry, {
+        token: session.token,
+        requestingUserId: session.userId,
+        owningOrgId: session.owningOrgId,
+        engagementId,
+        skip: (page - 1) * count,
         count,
-        sort,
-        order,
-        filter,
-      },
-    });
+      });
+
+      const products = await productQuery.run();
+
+      result.items = await Promise.all(
+        products.map(async (product) => this.readOne(product.id, session))
+      );
+      result.total = result.items.length;
+      // console.log(result.items);
+    } else {
+      result = await this.db.list<Product>({
+        session,
+        nodevar: 'product',
+        aclReadProp: 'canReadProducts',
+        aclEditProp: 'canCreateProduct',
+        props: [
+          { name: 'type', secure: false },
+          { name: 'books', secure: false, list: true },
+          { name: 'mediums', secure: false, list: true },
+          { name: 'purposes', secure: false, list: true },
+          { name: 'approach', secure: false },
+          { name: 'methodology', secure: false },
+        ],
+        input: {
+          page,
+          count,
+          sort,
+          order,
+          filter,
+        },
+      });
```

Comment:
We need to not use the old `database.service.ts` file functions. Instead, use the `query.helpers.ts` exports.